From f8e2429fa558961a6372282f82f61362c7e22356 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 19 Dec 2019 14:22:32 +0800
Subject: [PATCH 7/7] media: Enable mojo media when using v4l2 codec on desktop
 linux

The newest chromium turns to use mojo video decoder instead of gpu video
decoder.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 media/media_options.gni | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/media/media_options.gni b/media/media_options.gni
index 08fe4b174..e32f78816 100644
--- a/media/media_options.gni
+++ b/media/media_options.gni
@@ -154,7 +154,7 @@ declare_args() {
   enable_mojo_media =
       !is_fuchsia &&
       (is_android || is_chromeos || is_mac || is_win || enable_library_cdms ||
-       (is_desktop_linux && use_vaapi) || is_chromecast)
+       (is_desktop_linux && (use_vaapi || use_v4l2_codec)) || is_chromecast)
 
   # Enable the TestMojoMediaClient to be used in mojo MediaService. This is for
   # testing only and will override the default platform MojoMediaClient, if any.
@@ -219,7 +219,7 @@ if (enable_mojo_media) {
     ]
     _default_mojo_media_host = "gpu"
   } else if (is_chromeos || is_mac || is_win ||
-             (is_desktop_linux && use_vaapi)) {
+             (is_desktop_linux && (use_vaapi || use_v4l2_codec))) {
     _default_mojo_media_services = [ "video_decoder" ]
     _default_mojo_media_host = "gpu"
   }
-- 
2.11.0

